operators/multiclusterobservability/controllers/placementrule/placementrule_controller.go

```diff
@@ -514,44 +516,74 @@ func updateManagedClusterList(obj client.Object) {
 	}
 }
 
+// Do not reconcile objects if this instance of mch has the
+// `disableAddonAutomaticInstallationAnnotationKey` annotation
+func isAutomaticAddonInstallationDisabled(obj client.Object) bool {
+	annotations := obj.GetAnnotations()
+	if val, ok := annotations[disableAddonAutomaticInstallationAnnotationKey]; ok && strings.EqualFold(val, "true") {
```

Comment:
need to check whether annotations is nil

Reply:
it looks like golang will not panic when indexing a nil map. I can still add the check for code clarity, but it seems Go only panics if you try to insert into a nil map